src/pc/gfx/gfx_sdl2.c

```diff
@@ -163,49 +241,56 @@ static void gfx_sdl_init(void) {
     "Super Mario 64 PC port (OpenGL_ES2)";
     #endif
 
-    wnd = SDL_CreateWindow(
+    gContext.wnd = SDL_CreateWindow(
         window_title,
         configWindow.x, configWindow.y, configWindow.w, configWindow.h,
         SDL_WINDOW_OPENGL | SDL_WINDOW_SHOWN | SDL_WINDOW_RESIZABLE
     );
-    ctx = SDL_GL_CreateContext(wnd);
-
-    SDL_GL_SetSwapInterval(configWindow.vsync);
+    gContext.ctx = SDL_GL_CreateContext(gContext.wnd);
 
+    gfx_sdl_update_refresh_rate();
+    gfx_sdl_update_vsync();
     gfx_sdl_set_fullscreen();
 
     for (size_t i = 0; i < sizeof(windows_scancode_table) / sizeof(SDL_Scancode); i++) {
-        inverted_scancode_table[windows_scancode_table[i]] = i;
+        gContext.inverted_scancode_table[windows_scancode_table[i]] = i;
     }
 
     for (size_t i = 0; i < sizeof(scancode_rmapping_extended) / sizeof(scancode_rmapping_extended[0]); i++) {
-        inverted_scancode_table[scancode_rmapping_extended[i][0]] = inverted_scancode_table[scancode_rmapping_extended[i][1]] + 0x100;
+        gContext.inverted_scancode_table[scancode_rmapping_extended[i][0]] = gContext.inverted_scancode_table[scancode_rmapping_extended[i][1]] + 0x100;
     }
 
     for (size_t i = 0; i < sizeof(scancode_rmapping_nonextended) / sizeof(scancode_rmapping_nonextended[0]); i++) {
-        inverted_scancode_table[scancode_rmapping_nonextended[i][0]] = inverted_scancode_table[scancode_rmapping_nonextended[i][1]];
-        inverted_scancode_table[scancode_rmapping_nonextended[i][1]] += 0x100;
+        gContext.inverted_scancode_table[scancode_rmapping_nonextended[i][0]] = gContext.inverted_scancode_table[scancode_rmapping_nonextended[i][1]];
+        gContext.inverted_scancode_table[scancode_rmapping_nonextended[i][1]] += 0x100;
     }
 }
 
 static void gfx_sdl_main_loop(void (*run_one_game_iter)(void)) {
-    Uint32 t;
+    // Number of nanosecons a game's frame should take
+    const Uint32 FRAMETIME = 1e9 / FRAMERATE;
```

Comment:
Since you're using `SDL_GetPerformanceCounter()`, there's nothing that guarantees that it will be 1e9 ticks per second on any given platform. Might want to call `SDL_GetPerformanceFrequency()` on init and adjust the frametime value.